### PR TITLE
Rdpp 552 - Create second logfile for ADI

### PR DESCRIPTION
**Pull request by todanator**
JIRA: https://techweb.axway.com/jira/browse/RDPP-552
This change creates a second logfile that will be later consumed by arrow-cloud for use by the ADI analytics. 

Coincides with arrow PR: https://github.com/appcelerator/arrow/pull/246

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -136,7 +136,22 @@ function createHttpLogger(server, serviceType, options) {
...
+				req: bunyan.stdSerializers.req,
+				res: bunyan.stdSerializers.res,
+				route: routeSerializer
+			},
+			streams: [{
+				type: 'rotating-file',
+				period: '1d',
+				count: 1,
+				level: 'trace',
+				path: path.join(logDir, options && options.requestsLogFilename || 'adiAnalytics.log')
```
Comment (wickedest) [inline lib/logger.js]: That doesn't look right.  If `requestsLogFilename` is configured, then both `adiLogger` and `requestsLogger` should share the same log file.

Also, it's a bit persnickety but the name `adiAnalytics.log` does not conform with established file names in that directory - `requests-adi.log` or `adi-analytics.log` would be better names.

File: lib/logger.js [hunk 2]
```diff
@@ -189,13 +204,64 @@ function createHttpLogger(server, serviceType, options) {
...
 
 	/**
+	 * derive the port if its in the host string.
+	 * @param  {Object} req a express req object
+	 * @return {string}     string representation of port
+	 */
+	function getPort(req) {
+		var host = req.headers && req.headers.host;
+		var protocolSrc = null;
+		if (host && ((host.match(/:/g) || []).length) === 1) {
```
Comment (wickedest) [inline lib/logger.js]: If my URL was, `http://foo.com`, then the port would be `80` by default.

File: lib/logger.js [hunk 3]
```diff
@@ -206,6 +272,56 @@ function createHttpLogger(server, serviceType, options) {
...
+				type: 'transaction',
+				time: time,
+				path: uri,
+				protocol: req.protocol,
+				protocolSrc: port,
+				duration: responseTime,
+				status: getStatus(res),
+				serviceContexts: [
+					{
+						service: '',
```
Comment (wickedest) [inline lib/logger.js]: Maybe it's just the example you gave, but I noticed that `service`, etc. were `null`, but they are empty string (correct) here.

File: lib/logger.js [hunk 4]
```diff
@@ -206,6 +272,56 @@ function createHttpLogger(server, serviceType, options) {
...
+						duration: ''
+					}
+				],
+				customMsgAtts: {},
+				correlationId: getCorrelationId(req),
+				legs: [
+					{
+						uri: uri,
+						status: res.statusCode,
+						statusText: '',
```
Comment (wickedest) [inline lib/logger.js]: Maybe it's just the example you gave, but `statustext` was missing.  Also note capitalization differs from the spec (which _is_ inconsistent).

File: lib/logger.js [hunk 5]
```diff
@@ -206,6 +272,56 @@ function createHttpLogger(server, serviceType, options) {
...
+						method: req.method,
+						vhost: null,
+						wafStatus: 0,
+						bytesSent: res.get('Content-Length'),
```
Comment (wickedest) [inline lib/logger.js]: In your example, this was being written as a string, this should be `parseInt(res.get('Content-Length'), 10)`

File: lib/logger.js [hunk 6]
```diff
@@ -206,6 +272,56 @@ function createHttpLogger(server, serviceType, options) {
...
+						bytesReceived: req.get('Content-Length'),
+						remoteName: '',
+						remoteAddr: '',
+						localAddr: '',
```
Comment (wickedest) [inline lib/logger.js]: Missing `remotePort`

File: test/logger.js [hunk 7]
```diff
@@ -10,6 +10,16 @@ var should = require('should'),
 	request = require('request'),
 	tmpdir = path.join(require('os').tmpdir(), 'test-logger-' + Date.now());
 
+function readFile (filePath, cb) {
+	fs.readFile(filePath, 'utf8', function (err,data) {
+		if (err) {
+			cb(err);
```
Comment (wickedest) [inline test/logger.js]: The `cb` will be called twice if this errors.

File: lib/logger.js (hunk 4 above)
Comment (todanator) [inline lib/logger.js]: Yeah that is weird but whats whats in axway's spec for these transactions
https://docs.axway.com/u/documentation/api_gateway/7.5.2/webhelp_portal_admin/Content/AdminGuideTopics/log_event_settings.htm

File: lib/logger.js (hunk 1 above)
Comment (todanator) [inline lib/logger.js]: ooh good catch, oversight on my part.

File: lib/logger.js (hunk 2 above)
Comment (todanator) [inline lib/logger.js]: I'll add default 80.

File: lib/logger.js (hunk 3 above)
Comment (todanator) [inline lib/logger.js]: I think the example I uploaded was out of date. This is fixed in the current logs I'm making.

File: lib/logger.js (hunk 5 above)
Comment (todanator) [inline lib/logger.js]: Fixing

File: lib/logger.js (hunk 6 above)
Comment (todanator) [inline lib/logger.js]: added back

File: lib/logger.js [hunk 8]
```diff
@@ -303,8 +303,8 @@ function createHttpLogger(server, serviceType, options) {
 						method: req.method,
 						vhost: null,
 						wafStatus: 0,
-						bytesSent: res.get('Content-Length'),
-						bytesReceived: req.get('Content-Length'),
+						bytesSent: parseInt(res.get('Content-Length'), 10) || null,
```
Comment (wickedest) [inline lib/logger.js]: Nope: `parseInt('0', 10) || null` is `null`, not `0`

**Pull request merged**